src/models/model_loader.cpp

```diff
@@ -20,17 +24,58 @@ namespace ed
 namespace models
 {
 
+bool readSDFGeometry(tue::config::Reader r, geo::CompositeShapePtr& composite, std::stringstream& error, geo::Pose3D pose_offset = geo::Pose3D::identity())
+{
+    geo::Pose3D pose = geo::Pose3D::identity();
+    readPose(r, pose);
+    pose = pose_offset * pose;
+    if (r.readGroup("geometry"))
+    {
+        std::map<std::string, geo::ShapePtr> dummy_shape_cache;
+        geo::ShapePtr sub_shape = loadShape("", r, dummy_shape_cache, error);
+        if (sub_shape)
+        {
+            if (!composite)
+                composite.reset(new geo::CompositeShape);
+            composite->addShape(*sub_shape, pose);
+        }
+        r.endGroup();
+        return true;
+    }
+    else
+        return false;
+}
+
 // ----------------------------------------------------------------------------------------------------
 
 ModelLoader::ModelLoader()
 {
-    const char * mpath = ::getenv("ED_MODEL_PATH");
+    const char * edmpath = ::getenv("ED_MODEL_PATH");
+    if (edmpath)
+    {
+        std::stringstream ss(edmpath);
+        std::string item;
+        while (std::getline(ss, item, ':'))
+            if (!item.empty())
+                ed_model_paths_.push_back(item);
+    }
```

Comment:
Refactor into a function, and use it 3 times

Reply:
fixed